### PR TITLE
V22F-645 Cover all car controller code

### PR DESCRIPTION
**Pull request by KungFuDonkey**

### CODE REVIEW

**Review by shoekzema — CHANGES_REQUESTED** [review]
Missing documentation of CarControllerTests.cpp

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+{
+    float start = p_carController.GetCar()->ctrl.steer;
+    p_carController.SetSteerCmd(p_steer);
+    ASSERT_EQ(p_carController.GetSteerCmd(), p_steer + start);
+}
+
+/// @brief                 Tests the acceleration parameter of a car controller
+/// @param p_carController The car controller
+/// @param p_steer         The acceleration value
+void AccelTest(CarController p_carController, float p_accel)
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp:36]:
```suggestion
/// @param p_accel         The acceleration value
void AccelTest(CarController p_carController, float p_accel)
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+{
+    float start = p_carController.GetCar()->ctrl.accelCmd;
+    p_carController.SetAccelCmd(p_accel);
+    ASSERT_EQ(p_carController.GetAccelCmd(), p_accel + start);
+}
+
+/// @brief                 Tests the brake parameter of a car controller
+/// @param p_carController The car controller
+/// @param p_steer         The brake value
+void BrakeTest(CarController p_carController, float p_brake)
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp:46]:
```suggestion
/// @param p_brake         The brake value
void BrakeTest(CarController p_carController, float p_brake)
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+{
+    float start = p_carController.GetCar()->ctrl.brakeCmd;
+    p_carController.SetBrakeCmd(p_brake);
+    ASSERT_EQ(p_carController.GetBrakeCmd(), std::max(p_brake, start));
+}
+
+/// @brief                 Tests the clutch parameter of a car controller
+/// @param p_carController The car controller
+/// @param p_steer         The clutch value
+void ClutchTest(CarController p_carController, float p_clutch)
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp:56]:
```suggestion
/// @param p_clutch        The clutch value
void ClutchTest(CarController p_carController, float p_clutch)
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+{
+    p_carController.SetClutchCmd(p_clutch);
+    ASSERT_EQ(p_carController.GetClutchCmd(), p_clutch);
+}
+
+/// @brief                 Tests the lights parameter of a car controller
+/// @param p_carController The car controller
+/// @param p_steer         The lights value
+void LightTest(CarController p_carController, bool p_lights)
```
Comment (shoekzema) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp:65]:
```suggestion
/// @param p_lights        The lights value
void LightTest(CarController p_carController, bool p_lights)
```

**Review by shoekzema — CHANGES_REQUESTED** [review]

**Review by shoekzema — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+void ShowInterventionTest(InterventionAction p_action)
+{
+    IndicatorConfig::ClearInstance();
+    ASSERT_TRUE(SetupSingletonsFolder());
+    CarController carController;
+
+    auto activeBefore = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    carController.ShowIntervention(INTERVENTION_ACTION_BRAKE);
+
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/CarControllerTests.cpp:126]: I think this should be p_action
```suggestion

    carController.ShowIntervention(p_action);

```

**Review by noutheesink — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]

**Pull request merged**